coordinator/src/main/scala/filodb/coordinator/v2/FiloDbClusterDiscovery.scala

```diff
@@ -71,12 +71,22 @@ class FiloDbClusterDiscovery(settings: FilodbSettings,
   def shardsForLocalhost(numShards: Int): Seq[Int] = shardsForOrdinal(ordinalOfLocalhost, numShards)
 
   lazy private val hostNames = {
-    require(settings.minNumNodes.isDefined, "Minimum Number of Nodes config not provided")
+    require(settings.minNumNodes.isDefined, "[ClusterV2] Minimum Number of Nodes config not provided")
     if (settings.k8sHostFormat.isDefined) {
-      (0 until settings.minNumNodes.get).map(i => String.format(settings.k8sHostFormat.get, i.toString))
+      // This is used in kubernetes setup. We read the host format config and resolve the IP address from each host
+      val hosts = (0 until settings.minNumNodes.get)
+        .map(i => {
+          val resolvedIp = InetAddress.getByName(String.format(settings.k8sHostFormat.get, i.toString))
+          // return resolvedIpAddress and port combination for akka communication
+          s"${resolvedIp.getHostAddress()}:${settings.akkaPort}"
```

Comment:
This is a lazy val and will be initialized only once. Can the IP address of hosts change for any reason when any pod is bounced due to a crash etc? If yes, then this change will not capture the new IP address.

Reply:
discussed offline on this. We are now using hostname for akka actor registration and getting actor ref to communicate between the pods.